framework/test/jest/unit/specs/modules/chain/synchronizer/fast_chain_switching_mechanism/fast_chain_switching_mechanism.spec.js

```diff
@@ -14,149 +14,866 @@
+		) {
+			expect(loggerMock.info).toHaveBeenCalledWith(
+				{ peerId, reason },
+				'Applying penalty to peer and restarting synchronizer',
+			);
+			expect(channelMock.invoke).toHaveBeenCalledWith('network:applyPenalty', {
+				peerId,
+				penalty: 100,
+			});
+			expect(channelMock.publish).toHaveBeenCalledWith('chain:processor:sync', {
+				block: receivedBlock,
+			});
+		}
+
+		function checkApplyPenaltyAndAbortIsCalled(peerId, err) {
+			expect(loggerMock.info).toHaveBeenCalledWith(
+				{ err, peerId, reason: err.reason },
+				'Applying penalty to peer and aborting synchronization mechanism',
+			);
+			expect(channelMock.invoke).toHaveBeenCalledWith('network:applyPenalty', {
+				peerId,
+				penalty: 100,
+			});
+		}
+
+		function checkIfAbortIsCalled(error) {
+			expect(loggerMock.info).toHaveBeenCalledWith(
+				{
+					err: error,
+					reason: error.reason,
+				},
+				`Aborting synchronization mechanism with reason: ${error.reason}`,
+			);
+		}
+
+		beforeEach(async () => {
+			aBlock = newBlock();
+			// blocksModule.init will check whether the genesisBlock in storage matches the genesisBlock in
+			// memory. The following mock fakes this to be true
+			when(storageMock.entities.Block.begin)
+				.calledWith('loader:checkMemTables')
+				.mockResolvedValue({ genesisBlock: genesisBlockDevnet });
+			when(storageMock.entities.Account.get)
+				.calledWith({ isDelegate: true }, { limit: null })
+				.mockResolvedValue([{ publicKey: 'aPublicKey' }]);
+			// blocksModule.init will load the last block from storage and store it in ._lastBlock variable. The following mock
+			// simulates the last block in storage. So the storage has 2 blocks, the genesis block + a new one.
+			const lastBlock = newBlock({ height: genesisBlockDevnet.height + 1 });
+			when(storageMock.entities.Block.get)
+				.calledWith({}, { sort: 'height:desc', limit: 1, extended: true })
+				.mockResolvedValue([lastBlock]);
+			// Same thing but for BFT module,as it doesn't use extended flag set to true
+			when(storageMock.entities.Block.get)
+				.calledWith({}, { sort: 'height:desc', limit: 1 })
+				.mockResolvedValue([lastBlock]);
+			// BFT loads blocks from storage and extracts their headers
+			when(storageMock.entities.Block.get)
+				.calledWith(
+					{
+						height_gte: genesisBlockDevnet.height,
+						height_lte: lastBlock.height,
+					},
+					{ limit: null, sort: 'height:asc' },
+				)
+				.mockResolvedValue([genesisBlockDevnet, lastBlock]);
+
+			// Simulate finalized height stored in ChainMeta table is 0
+			when(storageMock.entities.ChainMeta.getKey)
+				.calledWith('BFT.finalizedHeight')
+				.mockResolvedValue(0);
+			jest.spyOn(fastChainSwitchingMechanism, '_queryBlocks');
+			jest.spyOn(fastChainSwitchingMechanism, '_switchChain');
+			jest.spyOn(fastChainSwitchingMechanism, '_validateBlocks');
+			await blocksModule.init();
+			await bftModule.init();
+		});
 
-		let syncMechanism;
+		afterEach(() => {
+			jest.clearAllMocks();
```

Comment:
isn't this done by config? @yatki

Reply:
I did this to reset call count on mocks but I am still unable to do it. I will talk to @yatki to find a solution for this.